custom-prompt/custom-prompt.cc

```diff
@@ -163,6 +166,190 @@ void Interval::print_long(std::ostream& ostream) const
  *****************************************************************************/
 extern "C" long long unsigned get_active_wid(void);
 
+/******************************************************************************
+ * Store information about the current Git repository.
+ *****************************************************************************/
+class GitRepository
+{
+private:
+    C::git_repository* repo;
+    std::filesystem::path gitdir;
+    C::git_reference* ref;
+    C::git_oid const* oid;
+    std::string description;
+    bool detached;
+    bool bisecting, cherrypicking, merging, rebasing, reverting;
```

Comment:
#### Suggestion

Don’t use so many variables. Have just a string `state`. Append additional information (e.g. number of steps in case of rebase) to that string.

Reply:
Done.